cli/azd/cmd/pipeline.go

```diff
@@ -125,21 +129,24 @@ func newPipelineConfigAction(
 		manager: pipeline.NewPipelineManager(
 			azCli, azdCtx, env, flags.global, commandRunner, console, flags.PipelineManagerArgs,
 		),
-		azdCtx:        azdCtx,
-		env:           env,
-		console:       console,
-		commandRunner: commandRunner,
+		azdCtx:         azdCtx,
+		accountManager: accountManager,
+		env:            env,
+		console:        console,
+		commandRunner:  commandRunner,
 	}
 
 	return pca
 }
 
 // Run implements action interface
 func (p *pipelineConfigAction) Run(ctx context.Context) (*actions.ActionResult, error) {
-	if p.env.GetSubscriptionId() == "" {
-		return nil, errors.New(
-			"infrastructure has not been provisioned. Please run `azd provision`",
-		)
+	// We need to ensure the env is set up correctly for provisioning,
+	// i.e., AZURE_SUBSCRIPTION_ID and AZURE_LOCATION must be set,
+	// so that the variables are set automatically on CI.
+	err := provisioning.EnsureSubscriptionAndLocation(ctx, p.console, p.env, p.accountManager)
```

Comment:
@vhvb1989 I think this is the old behavior, and works for `bicep`. I'm not sure about the `terraform` case. There seems to be more environment variables needed -- what is the end-to-end flow there?